### ISSUE TITLE
Custom NER tagging

### ISSUE THREAD

**Issue by mainrs** [guillaume-be/rust-bert#56]
Hi! I am looking for tools to tag ingredients of recipes to allow me extract information out of them. Given the example `1/2 cup of vegetable oil` I'd like to use NER to tag the quantity, unit and ingredient:
```
quantity: 1/2
unut: cup
ingredient: vegetable oil
```

I saw that there is already a NER implementation. How do I go about customizing it to tag the way I want it to? I know that I'd need to train a model (which I'd probably do using python) but I need to be able to tag incoming phrases using rust.

Any help is appreciated!

**Comment by guillaume-be** [comment]
Hello @SirWindfield and thank you for your interest in the library!

The tags generated by the model are stored in the `config.json` that is loaded to create a `Config` object for the NER pipeline. One section of the config file for the pre-trained model is as follows:
```json
  "id2label": {
    "0": "O",
    "1": "B-MISC",
    "2": "I-MISC",
    "3": "B-PER",
    "4": "I-PER",
    "5": "B-ORG",
    "6": "I-ORG",
    "7": "B-LOC",
    "8": "I-LOC"
  }
```
This is where the model loads its labels and maps them to entity labels. In your example, it may look as follows:
```json
  "id2label": {
    "0": "O",
    "1": "B-QTY",
    "2": "I-QTY",
    "3": "B-UNIT",
    "4": "I-UNIT",
    "5": "B-INGREDIENT",
    "6": "I-INGREDIENT",
  }
```

You would typically not need to update the config yourself because as you correctly stated, you need to train a model to extract these entities. I would also highly recommend Python for the training, and a lot of resources are available to do so. Possibly the simplest is to use [SimpleTransformers](https://github.com/ThilinaRajapakse/simpletransformers) allowing to train a model in 4 to 5 lines of code.

An example for NER is available [here](https://github.com/ThilinaRajapakse/simpletransformers/blob/master/examples/named_entity_recognition/named_entity_recognition.py). As you can see you'd need to create your training data using 2 Pandas dataframes with `sentence_idx`, `word`, `label`.

As the model is built on Transformers, it should be right away compatible with this crate. The last thing you'll have to do is convert the Pytorch weight file to the C-arrays used by the crate. please have a look at any of the conversion scripts in `./utils` to convert your local file.

Please give it a try, let me know if you need more help at any step of the way.

**Comment by SirWindfield** [comment]
Thanks for the explanation. I am quite new to NER in general. I will try it the coming week and report back if I can't figure it out :smile:
